File: java/socketd/src/main/java/org/noear/socketd/transport/client/ClientConnectorBase.java
package org.noear.socketd.transport.client;

/**
 * 客户端连接器基类
 *
 * @author noear
 * @since 2.0
 */
public abstract class ClientConnectorBase<T extends ClientInternal> implements ClientConnector{
    protected final T client;
    public ClientConnectorBase(T client){
        this.client = client;
    }

    /**
     * 获取配置
     * */
    @Override
    public ClientConfig getConfig(){
        return client.getConfig();
    }


    /**
     * 是否支持自动重连
     */
    @Override
    public boolean autoReconnect() {
        return client.getConfig().isAutoReconnect();
    }
}

File: python/socketd/socketd/transport/client/ClientConnectorBase.py
from abc import ABC

from .Client import ClientInternal
from .ClientConfig import ClientConfig
from .ClientConnector import ClientConnector


class ClientConnectorBase(ClientConnector, ABC):

    def __init__(self, client: ClientInternal):
        self.client: ClientInternal = client

    def get_config(self) -> ClientConfig:
        return self.client.get_config()

    def auto_reconnect(self):
        return self.client.get_config().is_auto_reconnect()



